File: csharp/nth-prime/NthPrime.cs
public static class NthPrime
{
    public static int Prime(int nth)
    {
        if (nth < 1) throw new ArgumentOutOfRangeException(nameof(nth), "The nth prime number must be positive integer(start from 1).");
        int count = 0;
        int num = 1;
        while (count  < nth)
        {
            num++;
            if (IsPrime(num)) count++;
        }
        return num;
    }
    private static bool IsPrime(int number)
    {
        if (number <= 1) return false;
        if (number <= 3) return true;
        if (number % 2 == 0 || number % 3 == 0) return false;
        for (int i = 5; i * i <= number; i += 6)
        {
            if (number % i == 0 || number % (i + 2) == 0) return false;
        }
        return true;
    }
}

File: typescript/nth-prime/nth-prime.ts
export function nth(n: number): number {
  if (n < 1) {
    throw new Error('Prime is not possible');
  }
  const primes: number[] = [2];
  let candidate = 3;
  while (primes.length < n) {
    if (isPrime(candidate, primes)) {
      primes.push(candidate);
    }
    candidate += 2;
  }
  return primes[n - 1];
}

function isPrime(num: number, primes: number[]): boolean {
  const limit = Math.sqrt(num);
  for (const p of primes) {
    if (p > limit) break;
    if (num % p === 0) return false;
  }
  return true;
}
